### ISSUE TITLE
__call__ using ClassWrapper does not update wrapped metric

### ISSUE THREAD

**Issue by geometrikal** [Lightning-AI/metrics#1220]
## 🐛 Bug

Using __call__ with ClasswiseWrapper does not update the Metric within, such that subsequent calls to compute are only using the last batch input.

### To Reproduce

See code sample

#### Code sample

```py
import numpy as np
import torch
import torchmetrics

np.random.seed(42)

m1 = torchmetrics.Precision(num_classes=3, average=None)
m2 = torchmetrics.Precision(num_classes=3, average=None)
mcw1 = torchmetrics.ClasswiseWrapper(torchmetrics.Precision(num_classes=3, average=None))
mcw2 = torchmetrics.ClasswiseWrapper(torchmetrics.Precision(num_classes=3, average=None))

for i in range(2):
    print(f"Epoch: {i}")
    preds = torch.Tensor(np.random.uniform(0, 1, (100, 3)))
    targets = torch.Tensor(np.random.randint(0, 2, (100, 3))).int()
    m1(preds, targets)
    m2.update(preds, targets)
    mcw1(preds, targets)
    mcw2.update(preds, targets)
    print(f"Prec:             {m1.compute().numpy()}")
    print(f"Prec update:      {m2.compute().numpy()}")
    print(f"Prec wrap:        {np.asarray([v.numpy() for v in mcw1.compute().values()])}")
    print(f"Prec wrap update: {np.asarray([v.numpy() for v in mcw2.compute().values()])}")
```

Output is:

```
Epoch: 0
Prec:             [0.42857143 0.537037   0.509434  ]
Prec update:      [0.42857143 0.537037   0.509434  ]
Prec wrap:        [0.42857143 0.537037   0.509434  ]
Prec wrap update: [0.42857143 0.537037   0.509434  ]
Epoch: 1
Prec:             [0.46464646 0.5247525  0.48039216]
Prec update:      [0.46464646 0.5247525  0.48039216]
Prec wrap:        [0.5       0.5106383 0.4489796]
Prec wrap update: [0.46464646 0.5247525  0.48039216]
```

### Expected behavior

The values should be the same whether calling `update` or `__call__`

### Environment

- TorchMetrics version: 0.9.3
- Python & PyTorch Version (e.g., 1.0): 1.12.0
- Any other relevant information such as OS (e.g., Linux): Ubuntu 20.04

**Comment by geometrikal** [comment]
I see this is because `full_state_update` is set to `True` inside the `ClasswiseWrapper` class. Perhaps the value of `full_state_update` should be set to that of the wrapped metric?
